Test failure in DiskImageDetailsBuild.spec.ts

### Bug description

All tests suddenly started failing with the following error:

```
 FAIL  src/lib/disk-image/DiskImageDetailsBuild.spec.ts > Refreshes logs correctly
AssertionError: expected "spy" to be called 2 times, but got 4 times
 ❯ vi.waitFor.timeout src/lib/disk-image/DiskImageDetailsBuild.spec.ts:98:46
     96|   await vi.waitFor(
     97|     () => {
     98|       expect(bootcClient.loadLogsFromFolder).toHaveBeenCalledTimes(2);
       |                                              ^
     99|     }
```
The issue is that the 2s refresh timer on the page is triggering 3 times before the waitFor (called every 250ms) is triggered.

### Operating system

macOS

### Version

next (development version)

### Steps to reproduce

_No response_

### Relevant log output

```shell

```

### Additional context

_No response_

--- a/packages/frontend/src/lib/disk-image/DiskImageDetailsBuild.spec.ts
+++ b/packages/frontend/src/lib/disk-image/DiskImageDetailsBuild.spec.ts
@@ -16,7 +16,7 @@
  ***********************************************************************/
 
 import { render, screen, waitFor } from '@testing-library/svelte';
-import { vi, test, expect, beforeAll, beforeEach } from 'vitest';
+import { vi, test, expect, beforeAll, beforeEach, afterAll } from 'vitest';
 import DiskImageDetailsBuild from './DiskImageDetailsBuild.svelte';
 import { bootcClient } from '/@/api/client';
 import type { Subscriber } from '/@shared/src/messages/MessageProxy';
@@ -48,10 +48,15 @@ beforeAll(() => {
       };
     },
   });
+  vi.useFakeTimers();
 });
 
 beforeEach(() => {
   vi.clearAllMocks();
+});
+
+afterAll(() => {
+  vi.useRealTimers();
 });
 
 const mockLogs = `Build log line 1
@@ -93,13 +98,6 @@ test('Refreshes logs correctly', async () => {
   render(DiskImageDetailsBuild, { folder: '/empty/logs' });
 
   // verify we start refreshing logs
-  await vi.waitFor(
-    () => {
-      expect(bootcClient.loadLogsFromFolder).toHaveBeenCalledTimes(2);
-    },
-    {
-      timeout: 3_500,
-      interval: 250,
-    },
-  );
+  await vi.advanceTimersByTimeAsync(2500);
+  expect(bootcClient.loadLogsFromFolder).toHaveBeenCalledTimes(2);
 });

